Pika - add publish_hook and consume_hook

ItayGibel-helios opened the pull request.

ItayGibel-helios committed:
Pika - add publish_hook and consume_hook

nozik commented on opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py:
```diff
@@ -18,125 +18,125 @@
-    "opentelemetry-instrumentation-sqlite3==0.25b0",
-    "opentelemetry-instrumentation-urllib==0.25b0",
-    "opentelemetry-instrumentation-wsgi==0.25b0",
+    "opentelemetry-instrumentation-dbapi==0.25b1",
```
I wouldn't make these updates upstream

ItayGibel-helios replied:
this file is auto-generated

nozik commented on instrumentation/opentelemetry-instrumentation-pika/tests/test_utils.py:
```diff
@@ -284,3 +328,51 @@ def test_decorate_basic_publish_no_properties(
+        method = mock.MagicMock(spec=Basic.Deliver)
+        properties = mock.MagicMock()
+        mock_body = b"mock_body"
+        publish_hook = mock.MagicMock()
```
Can you add a test that publish hook is a real function and not a mock?

nozik commented on instrumentation/opentelemetry-instrumentation-pika/tests/test_utils.py:
```diff
@@ -211,6 +211,50 @@ def test_decorate_callback(
+        method = mock.MagicMock(spec=Basic.Deliver)
+        properties = mock.MagicMock()
+        mock_body = b"mock_body"
+        consume_hook = mock.MagicMock()
```
Same for consume hook - could you do that with a real function?

Commit:
default value is invalid, was renamed (#755)

* default value is invalid, was renamed

* Update CHANGELOG.md

Co-authored-by: Owais Lone <owais@users.noreply.github.com>

owais committed:
Re-enable CI generate check and generate bootstrap_gen (#757)

Co-authored-by: Diego Hurtado <ocelotl@users.noreply.github.com>

owais committed:
Prepare release v1.6.2 and 0.25b2 (#762)

adamantike committed:
opentelemetry-instrument: Fix documented options (#747)

oxeye-nikolay committed:
Bugfix/instrument basic publish in pika (#759)

owais committed:
Added asgiref depedency to opentelemetry-instrumentation-asgi (#765)

opentelemetry-instrumentaiton-asgi does not contain an instrumentor for
asgiref. Instead, it implements an ASGI middleware that other
instrumentors import and use. As such, other instrumentors can take a
dependency on this package without asgiref being installed locally. This
commit adds `asgiref` as a dependency to the
opentelemetry-instrumentation-asgi package so the middleware can
function properly when used by another instrumentation.

lzchen committed:
Consolidate instrumentation documentation in docstrings (#754)

oxeye-yuval committed:
Bugfix: Pika basicConsume context propagation (#766)

adamantike committed:
Refactor code using pyupgrade for Python 3.6 (#770)

* Refactor code using pyupgrade for Python 3.6

This diff is the result of applying the following command to the
project:

```shell
find . -type f -name "*.py" -exec pyupgrade --py36-plus '{}' +
```

* Simplify yield from

adamantike committed:
`falcon`: Drop support for Python 3.4 (#774)

Remove unwanted support for Python versions <3.6.
This integration mistakenly lists Python 3.4 support, because it was
merged in
https://github.com/open-telemetry/opentelemetry-python/pull/1039,
after the merge of
https://github.com/open-telemetry/opentelemetry-python/pull/1099, so the
latter didn't consider `falcon`.

Python 3.4 is broken nevertheless, as this integration already includes
f-strings and other `opentelemetry` dependencies, which require Python 3.6.

Fixes #772.

adamantike committed:
django: Fix carrier usage on ASGI requests (#767)

* django: Fix carrier usage on ASGI requests

For ASGI requests, we must use `request.scope` instead of
`request.META`. This is because `ASGIGetter` retrieves the `headers` key
from the carrier [0], which is only present in `request.scope`.

[0] https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py#L133

* Add unit tests
